calico-enterprise/operations/monitor/metrics/recommended-metrics.mdx

```diff
@@ -0,0 +1,574 @@
+| Example value                    | <code>{endpoint="metrics-port", instance="10.0.1.20:9091", job="felix-metrics-svc", namespace="calico-system", pod="calico-node-qzpkt", service="felix-metrics-svc"} 0</code> |
+| Explanation                      | Checks if there have been any iptables-save, iptables-restore, or ipset command errors in the past five minutes. Keeps track of what node is reporting which error. |
+| Threshold value recommendation   | Occasional errors are normal.  Investigate error counters that rapidly rise.   For this specific metric it is worth focusing on the metric that is spiking, and referencing that metric information. |
+| Threshold breach symptoms        | Dependent on the specific metric that is logging errors.     |
+| Threshold breach recommendations | If more than one metric is rising, check if all rising metrics are related to a specific calico-node.  If this is the case, then the issue is local to that calico-node.  Check calico-node logs. Check resource usage for the node and calico-node pod. If more than one metric is rising rapidly across all calico-nodes, then it is a cluster-wide issue and cluster health must be checked.  Check cluster resource usage, cluster networking/infrastructure health, and restart calico-nodes and calico-typha pods. |
+| Priority level                   | Recommended.                                                 |
+
+## Felix time-based metrics
+
+### Dataplane apply time quantile 0.5/0.9/0.99
+
+| Dataplane apply time quantile 0.5/0.9/0.99 |                                                              |
+| ------------------------------------------ | ------------------------------------------------------------ |
+| Metric                                     | <code>felix_int_dataplane_apply_time_seconds{quantile="0.5"}</code><br /><code>felix_int_dataplane_apply_time_seconds{quantile="0.9"}</code><br /><code>felix_int_dataplane_apply_time_seconds{quantile="0.99"}</code> |
+| Example value                              | <code>felix_int_dataplane_apply_time_seconds{quantile="0.5"}:felix_int_dataplane_apply_time_seconds{**endpoint**="metrics-port", **instance**="10.0.1.30:9091", **job**="felix-metrics-svc", **namespace**="calico-system", **pod**="calico-node-6pcqm", quantile="0.5", **service**="felix-metrics-svc"} 0.020859218</code> |
+| Explanation                                | Time in seconds that it took to apply a dataplane update, viewed at the median, 90th percentile and 99th percentile. |
+| Threshold value recommendation             | Thresholds will vary depending on cluster size and rate of churn.  It is recommended that a baseline be set to determine a normal threshold value. Examples of what has been seen in the field: <br />- 3-node test cluster averaging 100ms <br />- 1000-node 15x federated cluster averaging 30s (potentially larger at start of business day). |
```

Comment:
> 1000-node 15x federated cluster averaging 30

This is very high, even on such a large cluster, where di we get that number from?  At start of day, it can be that high but I don't think it averages that high.

Reply:
Leon will have to speak to this number...

Reply:
@fasaxc Leon says today: "I believe I got this info from Shaun a year or so ago when I wrote the initial doc on Felix metrics https://docs.google.com/document/d/1qBu0izBJrRjnmjy3cgO05OWvAK_P_eEm1SNw7CTvq5c/edit#heading=h.5iw8ovc3asoo

Reply:
Fixed.